internal/blocksync/reactor.go

```diff
@@ -297,33 +297,13 @@ func (bcR *Reactor) Receive(e p2p.Envelope) {
 // Handle messages from the poolReactor telling the reactor what to do.
 // NOTE: Don't sleep in the FOR_LOOP or otherwise slow it down!
 func (bcR *Reactor) poolRoutine(stateSynced bool) {
-	bcR.metrics.Syncing.Set(1)
-	defer bcR.metrics.Syncing.Set(0)
+	didProcessCh, trySyncTicker, statusUpdateTicker, switchToConsensusTicker, blocksSynced, chainID, state, lastHundred, lastRate, initialCommitHasExtensions := bcR.setupPoolRoutine()
 
-	trySyncTicker := time.NewTicker(trySyncIntervalMS * time.Millisecond)
+	defer bcR.metrics.Syncing.Set(0)
```

Comment:
Thanks for starting this, @faddat 
What if  all these deferred calls were moved to another function that tears down the pollRoutine and this call is deferred?

Reply:
hey I'd be very cool with that.